### PR TITLE
Class manager update

### PR DESCRIPTION
**Pull request by anthony81799**
New Features

Update classManager to pass new classManagerTests.  Add field types, method return types, method parameter lists, and relationship types.

### CODE REVIEW

**Review by bjmulli1 — APPROVED** [review]
Looks a lot better then it did. Seems like it will still need some work.

File: src/controller/CommandController.java [hunk 1]
```diff
@@ -54,8 +54,8 @@ public int removeClass(String className) {
 		return result;
 	}
 
-	public int addField(String className, String fieldName) {
-		int result = getModel().addFields(className, fieldName);
+	public int addField(String className, String fieldName, String type) {
+		int result = getModel().addFields(className, fieldName, type);
```
Comment (ryanhaas) [inline src/controller/CommandController.java:58]: Might want to swap name and type order for consistency. Design choice.

File: src/controller/CommandController.java [hunk 2]
```diff
@@ -68,29 +68,29 @@ public int removeField(String className, String fieldName) {
 		return result;
 	}
 
-	public int addMethod(String className, String methodName) {
-		int result = getModel().addMethods(className, methodName);
+	public int addMethod(String className, String methodName, String returnType, String params) {
```
Comment (ryanhaas) [inline src/controller/CommandController.java:71]: Same as field, may want to swap order of method input to match typical programming format. Design choice.

File: src/controller/CommandController.java [hunk 3]
```diff
@@ -68,29 +68,29 @@ public int removeField(String className, String fieldName) {
...
+	public int removeMethod(String className, String methodName, String params) {
+		int result = getModel().removeMethods(className, methodName, params);
 		if(result == 0)
 			notify("removeMethod", getModel());
 		return result;
 	}
 
-	public int addRelationship(String class1, String class2) {
-		int result = getModel().addRelationship(class1, class2);
+	public int addRelationship(String class1, String class2, String type) {
```
Comment (ryanhaas) [inline src/controller/CommandController.java:85]: Similar to field and method, might want to change order to 'class1 type class2'

File: src/model/UMLClass.java [hunk 4]
```diff
@@ -17,8 +17,8 @@
 	
 	private String name;
 	public static final String [] types = {"int", "double", "float", "short", "long", "boolean", "String"};
-	private ArrayList<String> fields;
-	private ArrayList<String> methods;
+	private HashMap<String, Field> fields;
+	private HashMap<String, Method> methods;
```
Comment (ryanhaas) [inline src/model/UMLClass.java:21]: Put a comment saying the key format for methods (name + params)

File: src/model/UMLClass.java [hunk 5]
```diff
@@ -89,7 +91,7 @@ public void setName(String name) {
 	 * @return - true if class has method
 	 */
 	public boolean hasMethod(String methodName) {
```
Comment (ryanhaas) [inline src/model/UMLClass.java:93]: This method should take in the parameters as well as the key is built off of the parameters.

**Review by ryanhaas — APPROVED** [review]
Looks good for the most part, possibly look into the format changes I commented about regarding ordering of parameters.

**Review by anthony81799 — COMMENTED** [review]
Resolved all the suggested changes will merge this for functionality then after I fix the tests I will make a new pull request.

**Pull request merged**